codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/util/CustomValidationExceptionUtil.kt

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy.util
+
+import software.amazon.smithy.model.shapes.MemberShape
+import software.amazon.smithy.rust.codegen.traits.ValidationMessageTrait
+
+/**
+ * Helper function to determine if this [MemberShape] is a validation message either explicitly with the
+ * @validationMessage trait or implicitly because it is named "message"
+ */
+fun MemberShape.isValidationMessage(): Boolean {
+    return this.hasTrait(ValidationMessageTrait.ID) || this.memberName == "message"
```

Comment:
Model defaults should be handled at the model transformation stage. This centralizes all model modifications in one place and makes them easier to debug and maintain.

The idea is that users may omit the `@validationMessage` trait for convenience, but codegen should work with a fully-specified model. Handle these "ease of use" defaults in transformModel so we don't need
to check for them throughout the codebase.